xls2xml/xls2xml/tsv2xml.py

```diff
@@ -33,40 +32,15 @@
 xslt_filename = args.xslt
 
 tsv_reader = TSVReader(tsv_filename, tsv_conf, tsv_conf_key)
-tsv_validator = MetadataValidator(tsv_schema)
-
-if not tsv_reader.is_valid():
-    print('TSV file ' + tsv_filename + ' does not contain required fields!', file=sys.stderr)
-    quit(1)
-
-headers = tsv_reader.get_headers()
-if not headers:
-    print('TSV file ' + tsv_filename + ' does not have header row!', file=sys.stderr)
-    quit(1)
-
-tags = {header : header_to_xml_tag(header) for header in headers}
-
-has_error = False
-input_xml_root = etree.Element(tsv_conf_key+"Set")
-for row in tsv_reader:
-    if tsv_validator.validate_data(row, tsv_conf_key):
-        element_root = etree.SubElement(input_xml_root, tsv_conf_key)
-        for header in headers:
-            child_node = etree.SubElement(element_root, tags[header])
-            child_node.text = str(row.get(header, ''))
-    else:
-        has_error = True
-
-if has_error:
-    print('Please fix above error at file ' + tsv_filename + '!', file=sys.stderr)
+rows = []
+has_validation_error = utils.extract_rows(tsv_reader, tsv_conf_key, tsv_schema, rows)
+if has_validation_error:
     quit(1)
 
-xslt_tree = etree.parse(xslt_filename)
-transform = etree.XSLT(xslt_tree)
-output_xml = transform(input_xml_root)
+input_xml_root = utils.rows_to_xml(rows, tsv_conf_key)
+output_xml = utils.transform_xml(input_xml_root, xslt_filename)
 
 with open(xml_filename, 'w') as xml_file:
-    xml_file.write("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n")
```

Comment:
am not sure about python can we have exception handling while opening any files for reading/writing ?

Reply:
Those file opening are wrapped in a `with` context so if there is exception, the context will be exited cleanly (and file will be closed). Furthermore I don't want to handle the exceptions here more than what Python does.